gui/src/granite/GraniteWizard.tsx

```diff
@@ -760,36 +670,39 @@ const WizardContent: React.FC = () => {
           setOllamaOutdated(isOllamaOutdated);
           setServerState(newServerState);
 
+          const modelStatusMap = new Map<string, ModelStatus>(Object.entries(data.statusByModel));
+          setStatusByModel(modelStatusMap);
+
           const newStepStatuses = data.wizardState.stepStatuses as boolean[];
           setStepStatuses((prevStatuses) => {
-            if (prevStatuses[OLLAMA_STEP] && !newStepStatuses[OLLAMA_STEP]) {
-              //Ollama was just uninstalled/downgraded, return to the Ollama step
-              setActiveStep(OLLAMA_STEP);
-            } else if (
-              newStepStatuses[OLLAMA_STEP] &&
-              !newStepStatuses[MODELS_STEP] &&
-              prevStatuses[OLLAMA_STEP] !== newStepStatuses[OLLAMA_STEP]
-            ) {
-              setActiveStep(MODELS_STEP);
+            const ollamaStepChanged = prevStatuses[OLLAMA_STEP] !== newStepStatuses[OLLAMA_STEP];
+            const modelsStepChanged = prevStatuses[MODELS_STEP] !== newStepStatuses[MODELS_STEP];
+
+            if (ollamaStepChanged) {
+              if (!newStepStatuses[OLLAMA_STEP]) {
+                setActiveStep(OLLAMA_STEP);
+              } else if (newStepStatuses[OLLAMA_STEP] && !newStepStatuses[MODELS_STEP]) {
+                setActiveStep(MODELS_STEP);
+              }
             }
             if (newStepStatuses[MODELS_STEP]) {
               setModelInstallationProgress(100);
               setModelInstallationStatus("complete");
-              if (
-                !newStepStatuses[FINAL_STEP] &&
-                prevStatuses[MODELS_STEP] !== newStepStatuses[MODELS_STEP] &&
-                !isOllamaOutdated
-              ) {
+              if (!newStepStatuses[FINAL_STEP] && modelsStepChanged && !isOllamaOutdated) {
                 setActiveStep(FINAL_STEP);
               }
+            } else if (modelsStepChanged && modelInstallationStatusRef.current === "complete") {
+              //Model installation was complete but the step is not active anymore
+              //Reset the progress and status to allow the user to start the installation again
```

Comment:
Spaces after //

This comment took me some time to figure out, suggest for the first line:

```
 // Model installation was complete, and then some model was uninstalled
```